src/property-filter/__tests__/property-filter-token-editor-grouped.test.tsx

```diff
@@ -0,0 +1,309 @@
+  getValueFormRenderer: () => null,
+  externalProperty,
+};
+
+const dateProperty: InternalFilteringProperty = {
+  propertyKey: 'date',
+  propertyLabel: 'Date',
+  groupValuesLabel: 'Date values',
+  operators: ['=', '!='],
+  defaultOperator: '=',
+  getValueFormatter: () => (value: null | Date) => (value ? format(value, 'yyyy-MM-dd') : ''),
+  getValueFormRenderer:
+    () =>
+    ({ value }) => (
+      <FormField>
+        <DatePicker value={value ? format(value, 'yyyy-MM-dd') : ''} openCalendarAriaLabel={openCalendarAriaLabel} />
+      </FormField>
+    ),
+  externalProperty,
+};
+
+const defaultProps: TokenEditorProps = {
+  supportsGroups: true,
+  asyncProps: {},
+  customGroupsText: [],
+  freeTextFiltering: {
+    disabled: false,
+    operators: [':', '!:'],
+    defaultOperator: ':',
+  },
+  filteringProperties: [nameProperty, dateProperty],
+  filteringOptions: [],
+  i18nStrings: {
+    editTokenHeader: 'Edit token',
+    propertyText: 'Property',
+    operatorText: 'Operator',
+    valueText: 'Value',
+    cancelActionText: 'Cancel',
+    applyActionText: 'Apply',
+    tokenEditorTokenActionsLabel: token => `Remove actions ${token.propertyLabel} ${token.operator} ${token.value}`,
+    tokenEditorTokenRemoveLabel: () => 'Remove filter',
+    tokenEditorTokenRemoveFromGroupLabel: () => 'Remove filter from group',
+    tokenEditorAddNewTokenLabel: 'Add new filter',
+    tokenEditorAddTokenActionsLabel: 'Add filter actions',
+    tokenEditorAddExistingTokenLabel: token =>
+      `Add filter ${token.propertyLabel} ${token.operator} ${token.value} to group`,
+  },
+  onSubmit: () => {},
+  onDismiss: () => {},
+  standaloneTokens: [],
+  onChangeStandalone: () => {},
+  tempGroup: [{ property: nameProperty, operator: '=', value: 'John' }],
+  onChangeTempGroup: () => {},
+};
+
+function renderComponent(props?: Partial<TokenEditorProps>) {
+  const { container } = render(<TokenEditor {...defaultProps} {...props} />);
+  return new InternalPropertyFilterEditorDropdownWrapper(container);
+}
+
+describe.each(['xs', 'default'] as const)('breakpoints = %s', breakpoint => {
+  const isNarrow = breakpoint === 'default';
+
+  function findRemoveAction(wrapper: InternalPropertyFilterEditorDropdownWrapper, index: number) {
+    wrapper.findTokenRemoveActions(index)!.openDropdown();
+    return isNarrow
+      ? wrapper.findTokenRemoveActions(index)!.findMainAction()!
+      : wrapper.findTokenRemoveActions(index)!.findItems()[0];
+  }
+  function findRemoveFromGroupAction(wrapper: InternalPropertyFilterEditorDropdownWrapper, index: number) {
+    wrapper.findTokenRemoveActions(index)!.openDropdown();
+    return wrapper.findTokenRemoveActions(index)!.findItems()[isNarrow ? 0 : 1];
+  }
+
+  beforeEach(() => {
+    jest.mocked(useContainerBreakpoints).mockReturnValue([breakpoint, () => {}]);
+  });
+
+  test.each([false, true])('renders token editor with a single property, supportsGroups=%s', supportsGroups => {
```

Comment:
does it make sense to run such tests in the loop with the condition below? I would suggest to split them up in this case

Reply:
I think it does. There are certain fields that are supported when the groups are enabled and when they are not. Enabling groups unblocks more elements and those are tested separately with dedicated test cases.